roles/masters/templates/root/bin/install-registry.sh.j2

```diff
@@ -1,48 +1,43 @@
 #!/bin/bash
+# This script is an example on how to deploy or redeploy registry on a standard installation.
+#
+# You can modify it depending on your needs, ansible won't replace it if it alreay exists.
+#
+# (meaning, if you want to get the last version, just delete it and re-run the playbook.)
 set -e
+set -x
 
-#delete registry to create new one
-#   oc delete dc docker-registry
-#   oc delete svc docker-registry
-#   for i in $(oc get pods --no-headers | awk '/registry/{print $1}'); do oc delete pod $i; done
+delete() {
+    oc delete -n default dc docker-registry
+    oc delete -n default svc docker-registry
+    oc get pods -n default --no-headers -l 'deploymentconfig=docker-registry' | awk '/^docker-registry-/{print $1}' | xargs -r -n1 oc delete pod -n default
+}
 
-# Legacy <=3.0.1, need to create ServiceAccount
-#   echo \
-#       '{"kind":"ServiceAccount","apiVersion":"v1","metadata":{"name":"registry"}}' \
-#       | oc create -f -
-#   oc edit scc privileged
-# Add a line under users with the user name system:serviceaccount:default:registry.
+verify() {
+    nb=$(oc get pods -n default --no-headers -l 'deploymentconfig=docker-registry' | wc -l)
 
-# Ensure nodes for the registry are schedulable
-#   oadm manage-node {{ inventory_hostname }} --schedulable=true
+    if [ $nb -ge 1 ]; then
+        echo "ERROR: there are already a registry pod running"
+        echo "Please delete current dc/svc/pods for registry or use the '-f' option"
+        exit 1
+    fi
+}
 
-#
-# Create the registry with local volume
-#
+create_simple_registry() {
+    oadm registry --create --service-account=registry \
+         --config=/etc/{{ masters_etc_config }}/master/admin.kubeconfig \
+         --credentials=/etc/{{ masters_etc_config }}/master/openshift-registry.kubeconfig \
+         --selector="region=infra"
```

Comment:
good point.
but maybe we should also put
```
openshift_registry_selector: region=infra
openshift_router_selector: region=infra
```
in roles/master/defaults/main.yml
Actually even <code>region=primary</code> as the hosts.template does not have the label infra
those openshift_*_selector variables are not used anywhere for now.